include/util/percent.hpp

```diff
@@ -4,6 +4,14 @@
 #include <atomic>
 #include <iostream>
 
+// For isatty()
+#ifdef WIN32
+#include <io.h>
+#else
```

Comment:
It's not a binary Windows vs rest situation.
I would ifdef this on the POSIX.1-2001 standard and provide an else #error always failing.